src/ui/components/common/content_actions/report_button.rs

```diff
@@ -0,0 +1,45 @@
+use crate::{
+  ui::components::common::icon::{Icon, IconType},
+  utils::types::{ContentActionType, ReportModalData, ReportModalNode},
+};
+use leptos::*;
+
+fn report_content(id: i32, content_type: ContentActionType, creator_actor_id: String) {
```

Comment:
This can be a `PostId` or `CommentId` right? It'd pry be best to do this either with an enum (I'm not sure if a simple union would work IE `PostId | CommentId`)

Something like 

```rust
enum PostOrCommentId {
  Post(PostId)
  Comment(CommentId)
}
```

Reply:
I ended up making an enum like you suggested:

```rust
pub enum ContentId {
    Post(i32),
    Comment(i32)
}
```

[I used an `i32` for this for similar reasons I described in another comment.](https://github.com/LemmyNet/lemmy-ui-leptos/pull/107#discussion_r1694362491)